integration-tests/tests/update_worker_set.rs

```diff
@@ -225,3 +226,183 @@ fn worker_set_can_be_initialized_and_then_automatically_updated_during_proof_con
 
     assert_eq!(new_worker_set, expected_new_worker_set);
 }
+
+#[test]
+fn worker_set_cannot_be_updated_again_while_pending_worker_is_not_yet_confirmed() {
```

Comment:
I have no idea what this test is supposed to do, it's way too complicated

Reply:
can you simplify if possible and add documentation to explain the reasoning and what the expected behaviour under test is?

Reply:
@cgorenflo 

Addressed here: https://github.com/axelarnetwork/axelar-amplifier/pull/235/commits/0e37b0b11fbb484b1d0cfb73034bd1f287230cb2